File: Sif/src/app/core/models/articleTag.ts
import { Article } from './article';
import { Tag } from './tag';

export class ArticleTag {
  constructor(article: Article, tag: Tag) {
    this.articleId = article.articleId;
    this.tagId = tag.tagId;
  }

  articleId: number;
  tagId: number;
}

File: Thor.Models/Database/ArticleTag.cs
using DTO = Thor.Models.Dto;

namespace Thor.Models.Database;

public class ArticleTag
{
  public ArticleTag() {}

  public ArticleTag(DTO.ArticleTag articleTag)
  {
    ArticleId = articleTag.ArticleId;
    TagId = articleTag.TagId;
  }

  public int ArticleId { get; set; }
  public int TagId { get; set; }
  public Article Article { get; set; }
  public Tag Tag { get; set; }
}